core/src/components/picker-column-internal/picker-column-internal.tsx

```diff
@@ -0,0 +1,249 @@
+   *
+   * @internal
+   */
+  @Prop() numericInput = false;
+
+  /**
+   * Emitted when the value has changed.
+   */
+  @Event() ionChange!: EventEmitter<PickerColumnItem>;
+
+  @Watch('value')
+  valueChange() {
+    this.scrollActiveItemIntoView();
+  }
+
+  /**
+   * Only setup scroll listeners
+   * when the picker is visible, otherwise
+   * the container will have a scroll
+   * height of 0px.
+   */
+  componentWillLoad() {
+    const visibleCallback = (entries: IntersectionObserverEntry[]) => {
+      const ev = entries[0];
+
+      if (ev.isIntersecting) {
+        this.scrollActiveItemIntoView();
+        this.initializeScrollListener();
+      } else {
+        if (this.destroyScrollListener) {
+          this.destroyScrollListener();
+          this.destroyScrollListener = undefined;
+        }
+      }
+    }
+    new IntersectionObserver(visibleCallback, { threshold: 0.01 }).observe(this.el);
+
+    const parentEl = this.el.closest('ion-picker-internal') as HTMLIonPickerInternalElement | null;
+    if (parentEl !== null) {
+      parentEl.addEventListener('ionInputModeChange', (ev: any) => this.inputModeChange(ev));
+    }
+  }
+
+  scrollActiveItemIntoView() {
+    const activeEl = this.activeItem;
+
+    if (activeEl) {
+      this.centerPickerItemInView(activeEl, false);
+
+      /**
+       * This is needed because the initial
+       * scrollActiveItemIntoView call fires before
+       * the scroll event listener is setup.
+       */
+      activeEl.classList.add(PICKER_COL_ACTIVE);
+    }
+  }
+
+  private centerPickerItemInView = (target: HTMLElement, smooth = true) => {
+    this.el.scroll({
+      // (Vertical offset from parent) - (three empty picker rows) + (half the height of the target to ensure the scroll triggers)
+      top: target.offsetTop - (3 * target.clientHeight) + (target.clientHeight / 2),
+      left: 0,
+      behavior: smooth ? 'smooth' : undefined
+    });
+  }
+
+  /**
+   * When ionInputModeChange is emitted, each column
+   * needs to check if it is the one being made available
+   * for text entry.
+   */
+  private inputModeChange = (ev: PickerInternalCustomEvent) => {
+    if (!this.numericInput) { return; }
+
+    const { inputMode, inputModeColumn } = ev.detail;
+
+    /**
+     * If inputModeColumn is undefined then this means
+     * all numericInput columns are being selected.
+     */
+    const isColumnActive = inputModeColumn === undefined || inputModeColumn === this.el;
+
+    if (!inputMode || !isColumnActive) {
+      this.isActive = false;
+      return;
+    }
+
+    this.isActive = true;
+  }
+
+  /**
+   * When the column scrolls, the component
+   * needs to determine which item is centered
+   * in the view and will emit an ionChange with
+   * the item object.
+   */
+  private initializeScrollListener = () => {
+    const { el } = this;
+
+    let timeout: any;
+    let activeEl: HTMLElement | null = this.activeItem;
+    const scrollCallback = () => {
+      raf(() => {
```

Comment:
The `raf` is a utility for `requestAnimationFrame`. This lets us throttle the scroll callbacks so we do not clobber the DOM. For context, browsers typically throttle `requestAnimationFrame` to ~60fps normally and ~30fps on low battery mode.

The `raf` utility in particular lets us run `requestAnimationFrame` while avoiding Angular's Zone.js (performance improvement).

The `setTimeout` is there because we want to run certain code once the scrolling stops, but there is no "scrollend" event on the web at the moment. A setTimeout of `250ms` seems to work pretty well, but I'm open to tweaking the value if it seems too quick/too long.